build/build.cake

```diff
@@ -0,0 +1,238 @@
+// Methods
+//////////////////////////////////////////////////////////////////////
+
+void VerifyHeaders(bool Replace)
+{
+    var header = FileReadText("header.txt") + "\r\n";
+    bool hasMissing = false;
+
+    Func<IFileSystemInfo, bool> exclude_objDir =
+        fileSystemInfo => !fileSystemInfo.Path.Segments.Contains("obj");
+
+    var files = GetFiles(baseDir + "/**/*.cs", exclude_objDir).Where(file => 
+    {
+        var path = file.ToString();
+        return !(path.EndsWith(".g.cs") || path.EndsWith(".i.cs") || System.IO.Path.GetFileName(path).Contains("TemporaryGeneratedFile"));
+    });
+
+    Information("\nChecking " + files.Count() + " file header(s)");
+    foreach(var file in files)
+    {
+        var oldContent = FileReadText(file);
+        var rgx = new Regex("^(//.*\r?\n|\r?\n)*");
+        var newContent = header + rgx.Replace(oldContent, "");
+
+        if(!newContent.Equals(oldContent, StringComparison.Ordinal))
+        {
+            if(Replace)
+            {
+                Information("\nUpdating " + file + " header...");
+                FileWriteText(file, newContent);
+            }
+            else
+            {
+                Error("\nWrong/missing header on " + file);
+                hasMissing = true;
+            }
+        }
+    }
+
+    if(!Replace && hasMissing)
+    {
+        throw new Exception("Please run '.\\build.ps1 -target=UpdateHeaders' and commit the changes.");
+    }
+}
+
+//////////////////////////////////////////////////////////////////////
+// Default Task
+//////////////////////////////////////////////////////////////////////
+
+Task("Clean")
+    .Description("Clean the output folder")
+    .Does(() =>
+{
+    if(DirectoryExists(binDir))
+    {
+        Information("\nCleaning Working Directory");
+        CleanDirectory(binDir);
+    }
+    else
+    {
+        CreateDirectory(binDir);
+    }
+});
+
+Task("Verify")
+    .Description("Run pre-build verifications")
+    .IsDependentOn("Clean")
+    .Does(() =>
+{
+    VerifyHeaders(false);
+});
+
+Task("Version")
+    .Description("Updates the version information in all Projects")
+    .IsDependentOn("Verify")
+    .Does(() =>
+{
+    Information("\nDownloading NerdBank GitVersioning...");
+    var installSettings = new NuGetInstallSettings {
+        ExcludeVersion  = true,
+        Prerelease = true,
+        Version = "2.0.37-beta",
+        OutputDirectory = toolsDir
+    };
+    
+    NuGetInstall(new []{"nerdbank.gitversioning"}, installSettings);
+
+    Information("\nRetrieving version...");
+    var results = StartPowershellFile(versionClient);
+    Version = results[1].Properties["NuGetPackageVersion"].Value.ToString();
+    Information("\nBuild Version: " + Version);
+});
+
+Task("Build")
+    .Description("Build all projects and get the assemblies")
+    .IsDependentOn("Version")
+    .Does(() =>
+{
+    Information("\nBuilding Solution");
+    var buildSettings = new MSBuildSettings
+    {
+        MaxCpuCount = 0
```

Comment:
Why is this needed?

Reply:
This is the same as the /m flag for multithreading.